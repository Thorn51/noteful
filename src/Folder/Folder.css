.folder_container {
  display: flex;
  flex-direction: row;
}

.folder_sidebar {
  flex: 1;
  border: 1px solid lightgray;
  padding: 15px;
}

.folder_content {
  flex: 3;
  padding: 15px;
}

.folder {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 50px;
  border: lightgray solid 1px;
  margin: 10px;
}

.note {
  margin: 10px;
  border: lightgray solid 1px;
}

.note_name {
  font-size: 1.5rem;
  margin: 10px;
}

.note_modification {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.modification_date {
  margin: 10px;
}

.note_delete {
  margin: 10px;
}
