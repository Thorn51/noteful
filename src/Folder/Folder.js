import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import "./Folder.css";
import NoteContext from "../NoteContext";

export default class Folder extends React.Component {
  static contextType = NoteContext;
  render() {
    const folderNotes = this.context.notes.filter(
      note => note.folderId === this.props.match.params.folderId
    );
    const notes = Object.keys(folderNotes).map(key => {
      return (
        <div key={folderNotes[key].id} className="note">
          <Link to={`/note/${folderNotes[key].id}`}>
            <h2 className="note_name">{folderNotes[key].name}</h2>
          </Link>
          <div className="note_modification">
            <p className="mod_date">
              Date modified on {""}
              {moment(folderNotes[key].modified).format("Do MMM YYYY")}
            </p>
            <button className="delete_note">Delete Note</button>
          </div>
        </div>
      );
    });
    return (
      <section className="notes_section">
        <div className="folder_notes">{notes}</div>
        <button className="add_note">Add Note</button>
      </section>
    );
  }
}
