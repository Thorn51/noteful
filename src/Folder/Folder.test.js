import React from "react";
import ReactDOM from "react-dom";
import Folder from "./Folder";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Folder", () => {
  it("renders without throwing errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Folder />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the component as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Folder />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
