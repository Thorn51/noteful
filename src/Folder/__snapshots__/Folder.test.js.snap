// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder renders the component as expected 1`] = `
<section
  className="notes_section"
>
  <div
    className="folder_notes"
  />
  <a
    href="/add-note"
    onClick={[Function]}
  >
    <button
      className="add_note"
    >
      Add Note
    </button>
  </a>
</section>
`;
